modules/apps/object/object-admin-rest-impl/src/main/java/com/liferay/object/admin/rest/internal/resource/v1_0/ObjectDefinitionResourceImpl.java

```diff
@@ -527,7 +538,11 @@ public ObjectDefinition putObjectDefinition(
 					_getObjectFolderId(
 						objectDefinition.
 							getObjectFolderExternalReferenceCode()),
-					0);
+					0,
+					ObjectDefinitionSettingUtil.toObjectDefinitionSettings(
+						contextUser.getCompanyId(), _groupLocalService,
+						objectDefinition.getObjectDefinitionSettings(),
+						_objectDefinitionSettingLocalService));
```

Comment:
I noticed that in `ObjectDefinitionLocalService.addOrUpdateSystemObjectDefinition()`, you're passing `Collections.emptyList()` when `objectDefinition == null`. However, in the update scenario, this property is being passed from the API to the service layer, allowing it to be persisted. 

Should this story account for unmodifiableSystemObject, or is this the expected behavior?

Reply:
It was said in the product sync that story is not applicable unmodifiable system objects.